### PR TITLE
Introduce support for the Oracle JDBC driver

### LINKED ISSUE #1658: Add jdbc support for Oracle
Hi,

it would really help advocate for Quarkus in my company if it was possible to use it with an Oracle database.

I didn't find any information on how to add support for a new jdbc driver and I'm pretty noob on this. I quickly had a look at the [jdbc extensions](https://github.com/quarkusio/quarkus/tree/master/extensions/jdbc) but I didn't find a common pattern.

If somebody could give me a hint or two on how to add support for the Oracle driver, I will be happy to contribute :)

### PR DESCRIPTION
**Pull request by Sanne**
Fixes #1658

### CODE REVIEW

**Comment by Sanne** [comment]
Funny anecdote:

```
AuthorDate: Thu Mar 14 12:06:12 2019 +0000
```

And it's not a mistake :)  Just over two years.

**Comment by maxandersen** [comment]
Docker login with a quarkusio account seems best way forward. Hope ci will cache the image so it doesn't take forever :)

**Comment by Sanne** [comment]
@maxandersen I agree but I'm not familiar with github actions and its handling of secrets. Would be great if someone else could tick that box.

**Comment by famod** [comment]
We should also make sure to not run those tests in forks. I think CI should be able to check for a certain secret key and if it isn't there (default case for forks) then it should not enable the tests.

**Comment by galderz** [comment]
> Not a blocker, but would also be nice to leverage @galderz 's oracle/graal#3179 to better optimise the output.

I'd expect it to be available as part of 21.1 (supposed to come out in April).

**Comment by Sanne** [comment]
I've added essential documentation and rebased.

We discussed that CI jobs can be enabled at later time, so the PR is ready now.

**Comment by Sanne** [comment]
Seems CI failed because it couldn't download the Oracle DB docker image.. as expected. Sorry I forgot to think of how to toggle this.

**Comment by Sanne** [comment]
Added ad8cf45 : should work now.

**Comment by Sanne** [comment]
CI seems to like it better now; I just need an approval to merge it.

**Review by yrodiere — COMMENTED** [review]
I think there are a few missing bits, albeit minor ones. Maybe that was on purpose, I'm not entirely sure.

Apart from that, it looks good to me.

**Comment by gastaldi** [comment]
Looks like CI is complaining with a `Fatal error: java.lang.OutOfMemoryError: Could not allocate an aligned heap chunk`, which I don't think it's related to this PR

File: extensions/jdbc/jdbc-oracle/runtime/src/main/java/io/quarkus/jdbc/oracle/runtime/graal/OracleAgroalConnectionConfigurer.java [hunk 1]
```diff
@@ -0,0 +1,22 @@
+package io.quarkus.jdbc.oracle.runtime.graal;
+
+import io.agroal.api.configuration.supplier.AgroalDataSourceConfigurationSupplier;
+import io.agroal.api.exceptionsorter.MySQLExceptionSorter;
+import io.quarkus.agroal.runtime.AgroalConnectionConfigurer;
+import io.quarkus.agroal.runtime.JdbcDriver;
+import io.quarkus.datasource.common.runtime.DatabaseKind;
+
+@JdbcDriver(DatabaseKind.MYSQL)
```
Comment (yrodiere) [inline extensions/jdbc/jdbc-oracle/runtime/src/main/java/io/quarkus/jdbc/oracle/runtime/graal/OracleAgroalConnectionConfigurer.java:9]: Hum.... something looks wrong here?

**Review by yrodiere — COMMENTED** [review]
It looks like a copy/paste error slipped through the cracks.

Also, ideally, a test for the SSL and ExceptionSorter bits would be great... Though I suppose it's harder without the dev-services.

File: extensions/jdbc/jdbc-oracle/runtime/src/main/java/io/quarkus/jdbc/oracle/runtime/graal/OracleAgroalConnectionConfigurer.java (hunk 1 above)
Comment (Sanne) [reply extensions/jdbc/jdbc-oracle/runtime/src/main/java/io/quarkus/jdbc/oracle/runtime/graal/OracleAgroalConnectionConfigurer.java:9]: oops, this whole class shouldn't exist. It's not used either - deleding. Thanks!

**Review by yrodiere — APPROVED** [review]
I created #16172 to address the follow-ups. Feel free to merge once CI passes.

**Pull request merged**